BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -32,6 +33,63 @@ public class RealSense2 : Camera, IDisposable
         private bool _pipelineRunning = false;
         private float _depthScale = 1.0f;
 
+        #region Filter
+        public class RealSense2Filter
+        {
+            private ProcessingBlock _filter;
+
+            public bool Enabled { get; set; }
+            public Sensor.SensorOptions Options { get => _filter.Options; }
+
+            public RealSense2Filter(ProcessingBlock filter)
+            {
+                _filter = filter;
+            }
+
+            internal VideoFrame ApplyFilter(VideoFrame frame)
+            {
+                if (_filter is DecimationFilter)
+                    return (_filter as DecimationFilter).ApplyFilter(frame);
+                else if (_filter is SpatialFilter)
+                    return (_filter as SpatialFilter).ApplyFilter(frame);
+                else if (_filter is TemporalFilter)
+                    return (_filter as TemporalFilter).ApplyFilter(frame);
+                else if (_filter is DisparityTransform)
+                    return (_filter as DisparityTransform).ApplyFilter(frame);
+
+                throw new NotImplementedException();
+            }
+        }
+
+        public RealSense2Filter DecimationFilter { get; } = new RealSense2Filter(new DecimationFilter());
+        public RealSense2Filter SpatialFilter { get; } = new RealSense2Filter(new SpatialFilter());
+        public RealSense2Filter TemporalFilter { get; } = new RealSense2Filter(new TemporalFilter());
+        public RealSense2Filter DepthToDisparityFilter { get; } = new RealSense2Filter(new DisparityTransform(true));
+        public RealSense2Filter DisparityToDepthFilter { get; } = new RealSense2Filter(new DisparityTransform(false));
+
+        private VideoFrame FilterFrame(VideoFrame frame)
+        {
+            VideoFrame filteredFrame = frame;
+
+            if (TemporalFilter.Enabled)
```

Comment:
Candy: move this check inside the Filter.ApplyFilter method and you can remove 5 lines in this method.